python/lsst/ts/electrometer/mock_server.py

```diff
@@ -0,0 +1,64 @@
+import argparse
+import logging
+import re
+import threading
+import queue
+
+from serial import Serial
+
+
+class ElectrometerSimulator:
+    def __init__(self, port):
+        self.port = port
+        self.log = logging.Logger(__name__)
+        self.buffer = queue.Queue(maxsize=50000)
+        self.command_calls = {"*IDN?": self.get_identification}
+        self.commands = [
+            re.compile(r"^(?P<cmd>\*IDN\?)$")
+        ]
+
+    def cmd_loop(self):
+        while True:
+            self.log.debug("reading message")
+            rec = self.serial.read_until(b"\r")
+            self.log.debug(f"{rec}")
+            decoded_rec = rec.decode().strip()
+            self.log.debug(f"{decoded_rec}")
+            for command in self.commands:
+                self.log.debug(f"{command}")
+                matched_command = command.match(decoded_rec)
+                self.log.debug(f"{matched_command}")
+                if matched_command:
+                    self.log.debug(f"Matched {decoded_rec}")
+                    command_group = matched_command.group('cmd')
+                    if command_group in self.command_calls:
+                        called_command = self.command_calls[command_group]
+                        self.serial.write(called_command().encode())
+
+    def start(self):
+        self.log.debug(f"Starting simulator on {self.port}")
+        self.serial = Serial(self.port)
+        self.log.debug(f"real port is {self.serial.portstr}")
+        thread = threading.Thread(target=self.cmd_loop)
+        thread.start()
+
+    def stop(self):
+        pass
+
+    def get_identification(self):
+        return "KEITHLEY INSRUMENTS INC.,MODEL 6517B,4096271,A13/700X\n"
+
+
+def main():
```

Comment:
Please do not put command-line code in library modules. If you need a bin script to run the simulator then add it to the `bin/` directory.

If this is a legit need then you can still easily put almost all the code here by making `main` a classmethod on the class and calling it from the bin script. (For example salobj.BaseCsc has an `amain` method -- the leading "a" to indicate that it is asynchronous).

Reply:
Yes, you're right, but the mock_server is still in early development and so I just put the main there for early development purposes.